Tribler/Core/APIImplementation/LaunchManyCore.py

```diff
@@ -709,6 +714,13 @@ def sesscb_states_callback(self, states_list):
                     hops = self.session.config.get_default_number_hops()
                     self.update_download_hops(download, hops)
 
+            # Check the peers of this download every five seconds and add them to the payout manager when
+            # this peer runs a Tribler instance
+            if self.state_cb_count % 5 == 0 and download.get_hops() == 0 and self.payout_manager:
+                for peer in download.get_peerlist():
+                    if 'Tribler' in peer["extended_version"]:
+                        self.payout_manager.update_peer(peer["id"].decode('hex'), infohash, peer["dtotal"])
```

Comment:
Also, there might be garbage in the peer id: please do some input validation here/handle the exceptions from invalid hex encoding.

Reply:
The peer ID is already encoded in hex format: https://github.com/Tribler/tribler/blob/next/Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py#L852

Reply:
Alright 👍